### PR TITLE
osm os

### PR DESCRIPTION
**Pull request by wangzhao0217**
- **Create osm_vs_os**
- **add comapre between osm and os**
- **Add files via upload**
- **cn aggregation and geometries**
- **remove coherent build outside of build.R, the code/corenet_build_OSM.R is left as unfinished**

### CODE REVIEW

File: osm_os/osm_cohesive_network_complex.geojson
[the remark this answers is not in the document]
Comment (wangzhao0217) [reply osm_os/osm_cohesive_network_complex.geojson:1]: yes, agree Robin!

File: R/get_orcp_cn.R [hunk 1]
```diff
@@ -550,4 +549,41 @@ find_orcp_path = function(orcp_city_boundary, cohesive_network_city_boundary, OS
     }
     
     return(orcp_city_boundary)
+}
+
+
+line_merge = function(cohesive_network_city_boundary, OS_combined_net_city_boundary, combined_net_city_boundary) {
+
+cohesive_network_city_boundary = sf::st_transform(cohesive_network_city_boundary, crs = 27700)
```
[the remark this answers is not in the document]
Comment (wangzhao0217) [reply R/get_orcp_cn.R:557]: I've checked the code and removed the relevant lines. The CRS was made consistent in the previous steps.

File: R/get_orcp_cn.R [hunk 2]
```diff
@@ -505,21 +504,21 @@ find_orcp_path = function(orcp_city_boundary, cohesive_network_city_boundary, OS
...
   
-        tryCatch({
-          combined_orcp_path_sf_filtered = combined_orcp_path_sf_filtered |> 
-          dplyr::select(-all_fastest_bicycle_go_dutch) |>
-          dplyr::rename(all_fastest_bicycle_go_dutch = mean_all_fastest_bicycle_go_dutch)
-        }, error = function(e) {
-          message(sprintf("Error renaming column: %s", e$message))
-          print(names(combined_orcp_path_sf_filtered))
-        })
+        # tryCatch({
```
[the remark this answers is not in the document]
Comment (wangzhao0217) [reply R/get_orcp_cn.R:507]: Removed

**Review by Robinlovelace — APPROVED** [review]

**Review by Robinlovelace — COMMENTED** [review]
:fire:

**Review by Robinlovelace — COMMENTED** [review]
Looks good to me.

**Review by Robinlovelace — APPROVED** [review]
Looks good, I'm convinced this is working, happy for this to be merged when you are @wangzhao0217

File: .devcontainer/postCreateCommand.sh [hunk 3]
```diff
@@ -24,6 +24,9 @@ command -v cargo
 # Install odjitter using cargo
 cargo install --git https://github.com/dabreegster/odjitter --rev 32fb58bf7f0d68afd3b76b88cf6b1272c5c66828
 
+sudo ln -s /root/.cargo/bin/rustc /usr/local/bin/rustc
```
Comment (Robinlovelace) [inline .devcontainer/postCreateCommand.sh:27]: Not needed if we switch to the `ghcr.io/geocompx/docker:rust` image, right?

**Review by Robinlovelace — CHANGES_REQUESTED** [review]
Minor tweaks requested, let's get this merged.

File: .devcontainer/devcontainer.json [hunk 4]
```diff
@@ -31,6 +31,7 @@
 	// "remoteUser": "root"
 
 	// Run the script install-additional-dependencies.sh:
-	"postCreateCommand": "bash .devcontainer/postCreateCommand.sh"
+	"postCreateCommand": "bash ./.devcontainer/postCreateCommand.sh"
```
Comment (Robinlovelace) [inline .devcontainer/devcontainer.json:34]: Good fix!

**Review by Robinlovelace — APPROVED** [review]

File: .devcontainer/postCreateCommand.sh [hunk 5]
```diff
@@ -50,9 +47,3 @@ sudo ln -s ~/.cargo/bin/odjitter /usr/local/bin/odjitter
 
 # Make sure there's a newline at the end of the script
 echo "Script execution completed successfully."
-
-ls -hal ~/.cargo/bin/odjitter
-
-sudo ln -s ~/.cargo/bin/odjitter /usr/local/bin/odjitter
```
Comment (Robinlovelace) [inline .devcontainer/postCreateCommand.sh:56]: We need these!

**Review by Robinlovelace — APPROVED** [review]

**Pull request merged**